torch/_inductor/codegen/cpp_wrapper_cpu.py

```diff
@@ -2403,14 +2394,21 @@ def val_to_arg_str(self, val, type_=None) -> str:
 
         return self.val_to_arg_str_for_prim_type(val, type_)
 
-    def create_tmp_raii_handle_var(self, base_handle):
-        if base_handle.startswith(("wrap_with_raii_handle_if_needed",)):
-            # wrap_with_raii_handle_if_needed creates a temp RAIIAtenTensorHandle, so we need to
-            # explicitly store it. Otherwise, it will be destroyed before the fallback kernel call.
+    def create_tmp_raii_handle_var(self, base_handle: str) -> tuple[str, str]:
+        if base_handle.startswith(
+            (
+                "borrow_arrayref_tensor_as_tensor(",
+                "copy_arrayref_tensor_to_tensor(",
+                "wrap_with_raii_handle_if_needed(",
+                "RAIIAtenTensorHandle(",
```

Comment:
This change fixes an issue with Optional[Tensor] args, which are represented in the C-shim as `AtenTensorHandle*`. There are some ways around that other than simply saving off the temp handle, but for now I'm sticking with saving off the handle as no worse than the previous behavior.